Raspberry Pi OS bookworm環境でのビルドエラー対応

<!--
***********************************************************************
*                           << NOTE >>                                *
*  This repository have been programmed in C++.                       *
*                                                                     *
*  Please do not create tickets for other programming languages here. *
*  If you will create tickets for other programming language,         *
*  please see below:                                                  *
*                                                                     *
*  - Python : https://github.com/OpenRTM/OpenRTM-aist-Python/issues   *
*  - Java   : https://github.com/OpenRTM/OpenRTM-aist-Java/issues     *
*                                                                     *
***********************************************************************
-->

**Describe the bug**
OpenRTM 2.0.2版リリースに向けてLinux環境でビルドした場合、Raspberry Pi OS bookworm環境のみビルドエラーになる。bullseys, Ubuntu20.04, 22.04環境でのビルドは問題ない。ビルドが通るようになるまで、4種類のエラー対応が必要。
- 最初にcold属性のエラーが出る
  ```
  OpenRTM-aist/src/lib/coil/common/coil/Logger.cpp:286:5: error: function might be candidate for attribute ‘cold’ [-Werror=suggest-attribute=cold]
  ```
  - これはbookworm環境のgccバージョンが12.2.0のためで、一番トップのCMakeLists.txtで定義している「-Wsuggest-attribute=cold」でのバージョンを12.9.9にすれば解消できる
- 上記エラー対応をすると、次に下記エラーが発生する
  ```
  /usr/include/omniORB4/stringtypes.h:570:27: error: iteration 1073741823 invokes undefined behavior [-Werror=aggressive-loop-optimizations] 
    570 |     return ElemT(pd_data[i],pd_rel); 
        | 
  ```
  - これはビルドオプション「-fno-aggressive-loop-optimizations」を追加すると解消する
- 上記エラー対応をすると、次に下記エラーが発生する
  ```
  /src/OpenRTM-aist/src/lib/rtm/CorbaPort.h:1220:11: error: inlining failed in call to ‘RTC::CorbaPort::CorbaProviderHolder::CorbaProviderHolder(const RTC::CorbaPort::CorbaProviderHolder&)’: --param max-inline-insns-single limit reached [-Werror=inline]
   1220 |     class CorbaProviderHolder
        |           ^~~~~~~~~~~~~~~~~~~
  ```
  - これはビルドオプション「--param max-inline-insns-single=1000」を追加すると解消する
- 上記エラー対応をすると、次に下記エラーが発生する
  ```
  /src/OpenRTM-aist/src/lib/rtm/Manager.h:2221:11: error: inlining failed in call to ‘RTC::Manager::FactoryPredicate::FactoryPredicate(const RTC::Manager::FactoryPredicate&)’: --param inline-unit-growth limit reached [-Werror=inline]
   2221 |     class FactoryPredicate
        |
  ```
  - これはビルドオプション「--param inline-unit-growth=1000」を追加すると解消する



**Reproducibility**
100%

**Environment**
 - OS: Raspberry Pi OS bookworm
 - OpenRTM-aist Version 2.0.2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,6 +114,9 @@ elseif(UNIX)
 	set(INSTALL_RTM_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/openrtm-${RTM_MAJOR_VERSION}.${RTM_MINOR_VERSION}/cmake)
 	set(INSTALL_RTM_ETC_DIR ${CMAKE_INSTALL_SYSCONFDIR})
 	set(INSTALL_RTM_SHARE_DIR ${CMAKE_INSTALL_DATAROOTDIR})
+	if(${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 12.0.0)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-aggressive-loop-optimizations --param max-inline-insns-single=1000 --param inline-unit-growth=1000")
+	endif()
 else()
 	set(INSTALL_RTM_SCRIPT_DIR lib/openrtm-${RTM_MAJOR_VERSION}.${RTM_MINOR_VERSION}/py_helper)
 	set(INSTALL_RTM_BIN_DIR bin)
@@ -685,7 +688,7 @@ function(openrtm_common_set_compile_options target)
 		-Wstrict-null-sentinel
 		-Wstrict-overflow=5
 		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},6.9.9>: -Wstringop-overflow=4>
-		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},11.9.9>: -Wsuggest-attribute=cold>
+		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},12.9.9>: -Wsuggest-attribute=cold>
 		-Wsuggest-attribute=format
 		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},7.9.9>: -Wsuggest-attribute=malloc>
 		$<$<VERSION_GREATER:${CMAKE_CXX_COMPILER_VERSION},4.9.9>: -Wsuggest-override>

